### ISSUE TITLE
[BUG] Titeuf Mega Compet: frezze during ingame cutsceen.

### ISSUE THREAD

**Issue by atomic83GitHub** [PCSX2/pcsx2#4093]
**Describe the bug**
The game Titeuf Mega Compet suffer of a frezze during an ingame cutsceen at the start of the game.

A patch is available but the problem remain even still and is worth a github post, btw, +3 cyclerate will also break at the same moment even with the patch.

**To Reproduce**
Load the game then start a new game, see the cutsceen until it black screen.

**Expected behavior**
The cutsceen should continue to work regardless.

**Emulation Settings**
Everything default, the game does double signal but works with MTVU regardless.

**System Info (please complete the following information):**
PCSX2 Revision: pcsx2-v1.7.0-dev-817-gc998a51f7-windows-x86
OS: Windows 10
CPU: I9 9900k
GPU: RTX 3070

**Logs and Dumps**
https://drive.google.com/file/d/1JXO9OJz9DiEfU3s3KpKR35eq9xK8hNGH/view?usp=sharing

**Comment by refractionpcsx2** [comment]
Thought i better update here so I don't forget in future.

Game basically clears the FINISH event from the CSR register, runs a VIF packet containing a GIF packet (which has a FINISH flag in it), waits for VIF to finish, checks if the Finish flag is set and moans that it's set and refuses to go any further.

TL;DR the game is kinda stupid, but it's going to need some tests to see what the PS2 does, it might just be a silly timing thing..

**Comment by refractionpcsx2** [comment]
Update: ok so, I've worked out what's going on after looking at the data it's sending to the GS. The FINISH signal shouldn't happen at this point at all, basically all the VIF is sending to the GS is this

000000000000000e1000000000008005  // GIF Tag, basically 5 QW, EOP (end packet) 1 register 5x, the register is E (speciified)
00000000000000500000000000043000  // BITBLTBUF (Transfer destination location/format)
00000000000000510000000000000000  // TRXPOS (transfer square in memory)
00000000000000520000010000000100 // TRXREG (height/width of the square)
00000000000000530000000000000001 // TRXDIR (Basically start the transfer)
00000000000000610000000000000000  // FINISH (Fire finish interrupt when finished)

Finish is described as: When all ongoing host->local transfers have finished and any other drawing has finished, this fires.

Thing is, the game has just told it to start a transfer, then makes sure it hasn't finished, which it hasn't, because it's only told it to set the transfer off.

Now fixing this could be a challenge, we might have to hack it a bit because we don't *really* handle this stuff on the PCSX2 side and doing cross-thread stuff could be pretty horrible, at least while GSDX is a plugin.

So long story short: it's fixable, kinda

**Comment by refractionpcsx2** [comment]
Okay so turns out it's a game bug,  according to the official routine, you do what they did above, then you wait for finish to be triggered, then you reverse the fifo and do your GS download, however they decided to do the vif packet above then wait until FINISH is NOT set (aka the GS is still busy), this is obviously a bug, so the patch has been adjusted to correct this mistake.
